libs/wingc/src/type_check/symbol_env.rs

```diff
@@ -135,6 +140,24 @@ impl SymbolEnv {
 			});
 		}
 
+		// if this is a variable, verify that it adheres to the phase rules (notice that "this" is bound
+		// to the phase of the class).
+		if let SymbolKind::Variable(v) = &kind {
+			if self.phase == Phase::Inflight
+				&& v.phase == Phase::Preflight
+				&& symbol.name != "this"
+				&& symbol.name != CLASS_INIT_NAME
+			{
```

Comment:
IMO checking for phase compatibility seems like more of a responsibility of the type checker, not of our symbol environment abstraction, so I think this logic might belong elsewhere - what do you think?

Reply:
Mmmm... not 100% sure I agree with you on this. As I see this the symbol environment is part of our type system and it makes sense that it maintains its invariants. The alternative is to actually add this check every time I define a symbol within an environment which doesn't make sense to duplicate throughout the type checker.

If that's okay, I'll leave this here for now.

Reply:
@Chriscbr let me know if you are okay with this.